### ISSUE TITLE
remove model renderable from sceneview

### ISSUE THREAD

**Issue by basitsaleemmb** [appoly/ARCore-Location#64]
hi!
how can i remove modelRenderable from scene when the model is clicked with some fading animation ?

**Comment by SheikhZayed** [comment]
detach the node to which your anchor is hooked to.

this will do what you need

**Comment by basitsaleemmb** [comment]
@SheikhZayed it will be great if you can provide me the code snippet.

**Comment by SheikhZayed** [comment]
``node.getScene().onRemoveChild(node.getParent())
node.setRenderable(null)
session.update()``

more info [here](https://github.com/google-ar/sceneform-android-sdk/issues/135)

**Comment by basitsaleemmb** [comment]
thnku @SheikhZayed 

 andyRenderable = finalAndy.get();
 base = new Node();
 base.setRenderable(andyRenderable); 


i am adding base(NODE) to scene and it does not have any parent.

**Comment by SheikhZayed** [comment]
can you try this 

> private void removePreviousAnchors() { List<Node> nodeList = new ArrayList<>(arFragment.getArSceneView().getScene().getChildren()); for (Node childNode : nodeList) { if (childNode instanceof AnchorNode) { if (((AnchorNode) childNode).getAnchor() != null) { ((AnchorNode) childNode).getAnchor().detach(); ((AnchorNode) childNode).setParent(null); } } } }

**Comment by basitsaleemmb** [comment]
@SheikhZayed 

   andyRenderable = finalAndy.get();
   node= new Node();
   node.setRenderable(andyRenderable);

   node.setOnTapListener((v, event) -> {
                                        Toast.makeText(
                                                c, "I want to remove node from scene", Toast.LENGTH_LONG)
                                                .show();
                                  });

i want to remove node from scene on tap of node

**Comment by basitsaleemmb** [comment]
@SheikhZayed i think this way it can be done :)

public void removeMarker(LocationMarker locationMarker) {
            locationMarker.anchorNode.getAnchor().detach();
            locationMarker.anchorNode.setEnabled(false);
            locationMarker.anchorNode = null;

           locationScene.mLocationMarkers.remove(locationMarker);
    }
